charms/worker/k8s/src/charm.py

```diff
@@ -587,11 +589,7 @@ def _update_status(self):
         if self.is_dying:
             status.add(ops.WaitingStatus("Preparing to leave cluster"))
             return
-        if self.is_worker:
-            if not self.model.get_relation("cluster"):
-                status.add(ops.BlockedStatus("Missing cluster integration"))
-                assert False, "Missing cluster integration"  # nosec
-        else:
+        if self.is_control_plane:
```

Comment:
Cool -- this worker status is now handled in the `join_cluster` method.  We don't need it here anymore.